src/auth0/cache/mod.rs

```diff
@@ -29,8 +29,17 @@ pub trait Cache: Send + Sync + std::fmt::Debug {
     async fn get_token(&self, client_id: &str, aud: &str) -> Result<Option<Token>, CacheError>;
 
     async fn put_token(&self, client_id: &str, aud: &str, token: &Token) -> Result<(), CacheError>;
-}
 
-pub(in crate::auth0::cache) fn token_key(caller: &str, audience: &str) -> String {
-    format!("{}:{}:{}:{}", TOKEN_PREFIX, caller, TOKEN_VERSION, audience)
+    fn service_name(&self) -> &str;
```

Comment:
I don't think it makes sense to have these methods impld on Cache. Take a look at InMemoryCache, it doesn't need this at all and just complicates the constructor. The public API of cache shouldn't expose these methods

Reply:
I'll try to find a better way

Reply:
I removed the Cache changes first, but `service_name` is still in the implementations' struct since it needs to be passed to `token_key`